cpp/src/io/orc/stripe_enc.cu

```diff
@@ -1323,28 +1332,34 @@ void CompressOrcDataStreams(uint8_t* compressed_data,
                                                                             comp_stat,
                                                                             compressed_data,
                                                                             comp_blk_size,
-                                                                            max_comp_blk_size);
+                                                                            max_comp_blk_size,
+                                                                            block_align);
 
   if (compression == SNAPPY) {
     try {
-      if (detail::nvcomp_integration::is_stable_enabled()) {
+      if (nvcomp::is_compression_enabled(nvcomp::compression_type::SNAPPY)) {
         nvcomp::batched_compress(
-          nvcomp::compression_type::SNAPPY, comp_in, comp_out, comp_stat, comp_blk_size, stream);
+          nvcomp::compression_type::SNAPPY, comp_in, comp_out, comp_stat, stream);
       } else {
         gpu_snap(comp_in, comp_out, comp_stat, stream);
       }
     } catch (...) {
       // There was an error in compressing so set an error status for each block
-      thrust::for_each(rmm::exec_policy(stream),
-                       comp_stat.begin(),
-                       comp_stat.end(),
-                       [] __device__(decompress_status & stat) { stat.status = 1; });
+      thrust::for_each(
+        rmm::exec_policy(stream),
+        comp_stat.begin(),
+        comp_stat.end(),
+        [] __device__(compression_result & stat) { stat.status = compression_status::FAILURE; });
       // Since SNAPPY is the default compression (may not be explicitly requested), fall back to
       // writing without compression
     }
-  } else if (compression == ZLIB and detail::nvcomp_integration::is_all_enabled()) {
+  } else if (compression == ZLIB and
+             nvcomp::is_compression_enabled(nvcomp::compression_type::DEFLATE)) {
     nvcomp::batched_compress(
-      nvcomp::compression_type::DEFLATE, comp_in, comp_out, comp_stat, comp_blk_size, stream);
+      nvcomp::compression_type::DEFLATE, comp_in, comp_out, comp_stat, stream);
+  } else if (compression == ZSTD and
+             nvcomp::is_compression_enabled(nvcomp::compression_type::ZSTD)) {
+    nvcomp::batched_compress(nvcomp::compression_type::ZSTD, comp_in, comp_out, comp_stat, stream);
```

Comment:
It's the only one that we made "optional", because it's the default. The others are opt-in, so if they fail we'll fail the write. 
I'll iron out/expand this logic in a separate PR. One of the useful features would be to have an env var to configure if we allow fall back to uncompressed output.